pkg/skaffold/initializer/build/resolve.go

```diff
@@ -74,14 +83,37 @@ func (d *defaultBuildInitializer) resolveBuilderImagesInteractively() error {
 		}
 
 		if choice != NoBuilder {
-			pairs = append(pairs, BuilderImagePair{Builder: choiceMap[choice], ImageName: image})
+			d.builderImagePairs = append(d.builderImagePairs, BuilderImagePair{Builder: choiceMap[choice], ImageName: image})
 			choices = util.RemoveFromSlice(choices, choice)
 		}
 		d.unresolvedImages = util.RemoveFromSlice(d.unresolvedImages, image)
 	}
 	if len(choices) > 0 {
-		logrus.Warnf("unused builder configs found in repository: %v", choices)
+		// TODO(nkubala): should we ask user if they want to generate here?
+		for _, choice := range choices {
+			d.generatedBuilderImagePairs = append(d.generatedBuilderImagePairs, getGeneratedBuilderPair(choiceMap[choice]))
+		}
 	}
-	d.builderImagePairs = append(d.builderImagePairs, pairs...)
 	return nil
 }
+
+func getGeneratedBuilderPair(b InitBuilder) GeneratedBuilderImagePair {
```

Comment:
Can we add some tests for this method?
`filepath.Dir()` will return all but last component meaning it will contain `\`, `.` which are not valid image name chars

https://play.golang.org/p/huk3EmORFw5

Reply:
this is implicitly tested through `TestResolveBuilderImages` and `TestAutoSelectBuilders`. because of the way we detect builders in the `analyze` step of the initializer, the last 5 cases in your linked snippet will never happen.

Reply:
I am not sure if this true. 
I run checked out your branch and ran into this issue

Project tree:
```
examples/getting-started <say this is root of the repo>
|__app
      |__ deploy
              |---- Dockerfile
      |- main.go
```
Currently, the code assumes you are running `skaffold init` from inside `examples/app`. This logic is very sensitive to run context.
```

tejaldesai@@helm-deployment (init-k8sgen)$ ../../out/skaffold init --XXenableManifestGeneration
adding manifest path app/deploy/deployment.yaml for image app/deploy
apiVersion: skaffold/v2alpha4
kind: Config
metadata:
  name: helm-deployment
deploy:
  kubectl:
    manifests:
    - app/deploy/deployment.yaml

app/deploy/deployment.yaml - apiVersion: v1
kind: Service
metadata:
  name: app/deploy
  labels:
    app: app/deploy
spec:
  clusterIP: None
  selector:
    app: app/deploy
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: app/deploy
  labels:
    app: app/deploy
spec:
  replicas: 1
  selector:
    matchLabels:
      app: app/deploy
  template:
    metadata:
      labels:
        app: app/deploy
    spec:
      containers:
      - name: app/deploy
        image: app/deploy

Do you want to write this configuration, along with the generated k8s manifests, to skaffold.yaml? [y/n]: y
Generated manifest app/deploy/deployment.yaml was written
Configuration skaffold.yaml was written
You can now run [skaffold build] to build the artifacts
or [skaffold run] to build and deploy
or [skaffold dev] to enter development mode, with auto-redeploy

````

Reply:
so I dug into this one more, and `skaffold init` actually doesn't even work with this directory structure :/

i'm going to add sanitization to this PR, and open an issue to fix this separately.